### DISCUSSION TITLE
App crash EXC_BAD_ACCESS if Lottie is part of dependent package and is used in parent project as well

### DISCUSSION THREAD

**Discussion by ikaransingla** [airbnb/lottie-ios#1866] [General]
Lottie was working fine in my project after updating AnimationView to LottieAnimationView even in XIBs too.
It's been crashing since the time I have integrated an SPM package which also has Lottie as the dependency.
Both the project and the package has the same version of Lottie (4.0.1). Tried downgrading the version to 3.5.0 as well but it's still crashing.

## Which Version of Lottie are you using?

3.5.0, 4.1.0

## Expected Behavior 
The animation should work fine.

## Actual Behavior
Facing crash at the time of animation assignment. 
![Screenshot 2022-12-17 at 10 18 32 PM](https://user-images.githubusercontent.com/37956337/208252445-5fde6279-0273-465f-9bf5-4d90b0764999.png)

The animation is working fine inside the added package though.

## Animation JSON
Facing this crash for all json files. Attaching a sample one for reference
https://assets10.lottiefiles.com/datafiles/KddzycchL2y14pT/data.json

**Comment by calda** [comment]
Can you share a sample project demonstrating this issue?

**Comment by ikaransingla** [comment]
I've created a sample project which also have the same issue. 

[LottieTest.zip](https://github.com/airbnb/lottie-ios/files/10253355/LottieTest.zip)

The package used in this project is at https://github.com/ikaransingla/LottiePackage

The code used to build the framework is 
[LottiePackageTest.zip](https://github.com/airbnb/lottie-ios/files/10253356/LottiePackageTest.zip)

**Comment by calda** [comment]
I think having multiple copies of the same library loaded / compiled separately is not very well-supported. I know storyboards work by expecting there to be a unique `Lottie.LottieAnimationView` type, which isn't the case in this project setup. Since this seems to be an SPM / linker / Xcode / storyboard problem and not a bug with Lottie itself I've converted this from an issue to a discussion.

**Reply by calda** [reply]
The fix here would be to update the package setup / dependency graph to only have a single copy of the Lottie module. If that's not feasible another workaround could be to avoid using Lottie via storyboards -- initializing your `LottieAnimationView` via code instead probably wouldn't have this EXC_BAD_ACCESS issue.

**Reply by calda** [reply]
These sorts of warnings are relevant here -- multiple copies of the `Lottie.LottieAnimationView` symbol are present so runtime behavior is undefined

> objc[12229]: Class _TtC6Lottie19LottieAnimationView is implemented in both /Users/cal/Library/Developer/Xcode/DerivedData/LottieTest-hhxynuzdyfdatkbpzrkaxtrxpigk/Build/Products/Debug-iphonesimulator/LottiePackageTest.framework/LottiePackageTest (0x10604d568) and /Users/cal/Library/Developer/CoreSimulator/Devices/FA923632-7372-4747-9A51-8B4024E513CD/data/Containers/Bundle/Application/6F3C7A78-80BB-4ABE-B365-715D812FC47F/LottieTest.app/LottieTest (0x104922a20). One of the two will be used. Which one is undefined.

**Comment by asadkazmi-mset** [comment]
@ikaransingla 
I'm facing this issue on v4.4.3, did you able to fix that issue on your end?

**Reply by jayakarthi-plotline** [reply]
Did you solve the issue?

**Comment by jayakarthi-plotline** [comment]
@ikaransingla were you able to solve this?
